### DISCUSSION TITLE
Support PostgreSQL `jsonb` columns

### DISCUSSION THREAD

**Discussion by oXtxNt9U** [spatie/laravel-schemaless-attributes#107] [Ideas]
PostgreSQL has [different data types](https://www.postgresql.org/docs/current/datatype-json.html) for dealing with JSON. Using schemaless attributes, the resulting column is always of type `json`. But in almost all cases one wants to use `jsonb` over `json` when working with Postgres.

I saw this previous issue where someone  asked something similar, but the answers given are a bit minimal:
https://github.com/spatie/laravel-schemaless-attributes/issues/15

I'm not sure if it's supposed to be supported already. If it's already possible or there's a workaround using this package please let me know! 

Otherwise, would it be possible to add support for this? 
E.g. something like `$table->schemalessAttributes('extra_attributes', [ 'type' => 'jsonb'])`.

Many thanks in advance.

**Comment by danielbehrendt** [comment]
I'm also interested in support of `jsonb` columns as I recently switched to Postgres for a new project. 

Is there anything speaking against `jsonb` type when using schemaless attributes?

**Comment by beholdr** [comment]
There is error in the Filament relations https://github.com/filamentphp/filament/issues/8698 when using `json` instead of `jsonb`. So it would be cool if this library could support `jsonb` too.

Currently as workaround I'm using a manual `jsonb` type for schemaless field :

```php
$table->jsonb('data')->nullable();
```
